File: regiondeployer/azureauthenticator.fs
module regiondeployer

open System.Configuration
open System.Linq
open Microsoft.Azure.Management.Fluent
open Microsoft.Azure.Management.ResourceManager.Fluent
open System
open MoonMachine.Infrastructure.Helpers
open MoonMachine.credentials

let public createauthenticatedgateway() : IAzure = 
    let globalconfig = ProjectCredentials.get.matchingcurrentenvironment()

    let gateway = (SdkContext.AzureCredentialsFactory
        .FromServicePrincipal(
            globalconfig.AzureClientId,
            globalconfig.AzureClientSecret,
            globalconfig.AzureTenantId,
            AzureEnvironment.AzureGlobalCloud
        )
    |> Azure.Authenticate)

    gateway.WithDefaultSubscription()
        

File: regiondeployer/AzureAuthenticator.cs
namespace RegionDeployer;

using Microsoft.Azure.Management.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent;
using regiondeployer;

public class AzureAuthenticator
{
    public IAzure CreateAuthenticatedGateway(ConfigFile config)
    {
        var credentials = SdkContext.AzureCredentialsFactory
            .FromServicePrincipal(
                config.AzureClientId,
                config.AzureClientSecret,
                config.AzureTenantId,
                AzureEnvironment.AzureGlobalCloud);

        var authenticatedAzure = Azure.Authenticate(credentials);
        return authenticatedAzure.WithDefaultSubscription();
    }
}
